daemon/src/setup_contract_actor.rs

```diff
@@ -127,11 +130,22 @@ pub fn new(
             .map(|(tx, _, digits)| (digits.range(), (tx, digits)))
             .collect::<HashMap<_, _>>();
 
+        let mut signed_lock_tx = wallet.sign(lock_tx).await.unwrap();
+        send_to_other(SetupMsg::Msg2(Msg2 {
+            signed_lock: signed_lock_tx.clone(),
+        }));
+        let msg2 = receiver.recv().await.unwrap().try_into_msg2().unwrap();
+        signed_lock_tx.merge(msg2.signed_lock).unwrap();
+
+        // TODO: In case we sign+send but never receive (the signed lock_tx from the other party) we
+        // need some fallback handling (after x time) to spend the outputs in a different way so the
+        // other party cannot hold us hostage
+
         Dlc {
             identity: sk,
             revocation: rev_sk,
             publish: publish_sk,
-            lock: lock_tx,
+            lock: signed_lock_tx,
```

Comment:
Can we just do `signed_lock_tx.extract_tx()` here and move from `Psbt` to `Transaction`?